src/blocks/Block.as

```diff
@@ -900,6 +900,13 @@ public class Block extends Sprite {
 		Scratch.app.runtime.interp.toggleThread(topBlock(), Scratch.app.viewedObj(), 1);
 	}
 
+	public function demo():void{
+		//make a test duplicate and exec
+		var b:Block = this.duplicate(false);
+		b.nextBlock = null;
+		Scratch.app.runtime.interp.toggleThread(b, Scratch.app.viewedObj(), 1, true);
```

Comment:
Have you tried `b.visible = false` here? It seems like that would eliminate the need for the `toggleThread` change and might help in other ways. For example, it might avoid double-rendering the block.